### PR TITLE
Rel1 39 split part4

### PR DESCRIPTION
**Pull request by mmeller-wikia**

### CODE REVIEW

**Review by Michal-Chatlas1221 — CHANGES_REQUESTED** [review]
Also, please give some description about the scope of the changes, and intentions behind those changes. It seems that a lot of the diff is not related to upgrade.

File: ServiceWiring.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use MediaWiki\MediaWikiServices;
+use Reverb\Notification\NotificationBroadcastFactory;
+use WikiDomain\WikiConfigDataService;
+
+return [
+	CheevosClient::class => static function ( MediaWikiServices $services ): CheevosClient {
+		$config = $services->getMainConfig();
+		return new CheevosClient(
+			$services->getService( SERVICE_HTTP_CLIENT ),
+			$config->get( 'CheevosHost' ),
```
[the remark this answers is not in the document]
Comment (mmeller-wikia) [reply ServiceWiring.php:19]: I will stay with current solution for consistency - our http client classes gets `$serviceUrl` directly

File: src/AchievementService.php
[the remark this answers is not in the document]
Comment (mmeller-wikia) [reply src/AchievementService.php:1]: `AchievementService` and `FriendService` are new, they should replace `Cheevos` class.
I cannot modify `Cheevos` class public interface as it is widely used outside Cheevos extension

File: ServiceWiring.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Michal-Chatlas1221) [reply ServiceWiring.php:19]: The comment I wrote didn't express my intention very well. My main concern is the headers param below, which would likely be better off being just the config field

**Comment by mmeller-wikia** [comment]
> Also, please give some description about the scope of the changes, and intentions behind those changes. It seems that a lot of the diff is not related to upgrade.

Added description to original PR https://github.com/Wikia/mediawiki-extensions-Cheevos/pull/10

File: .phan/config.php [hunk 2]
```diff
@@ -4,6 +4,9 @@
 
 $cfg['suppress_issue_types'] = [
```
[the remark this answers is not in the document]
Comment (Michal-Chatlas1221) [reply .phan/config.php:5]: "we don't have enough capacity to properly resolve them" that could be the comment

**Review by Michal-Chatlas1221 — APPROVED** [review]

**Pull request closed without merging**